crates/fluss/src/client/table/scanner.rs

```diff
@@ -232,6 +297,13 @@ impl LogFetcher {
         }
     }
 
+    fn is_projection_enabled(&self) -> bool {
+        self.projected_fields
```

Comment:
we can force it's projection_enabled when projected_fields is not none to make the logic simple? Throw exception in `pub fn project(mut self, column_indices: &[usize])` when column_indices is empty.